src/api/apiMethods.js

```diff
@@ -55,54 +56,162 @@ const getApiMethodsSelector = () =>
+      const _getCellGroupTypeFromColumnGroupIndex = (columnGroupIndex) => {
+        const totalColumnGroups =
+          fixedColumnGroups.length +
+          scrollableColumnGroups.length +
+          fixedRightColumnGroups.length;
+
+        if (
+          columnGroupIndex < 0 ||
+          columnGroupIndex > totalColumnGroups ||
+          !Number.isInteger(columnGroupIndex)
+        ) {
+          throw `columnGroupIndex must be an integer between 0 and ${
+            totalColumnGroups - 1
+          } inclusive`;
+        }
+
+        if (columnGroupIndex < fixedColumnGroups.length) {
+          return CellGroupType.FIXED;
+        }
+
+        if (
+          columnGroupIndex <
+          fixedColumnGroups.length + scrollableColumnGroups.length
+        ) {
+          return CellGroupType.SCROLLABLE;
+        }
+
+        return CellGroupType.FIXED_RIGHT;
+      };
+
+      const _getColumnLocalIndex = (columnIndex, cellGroupType) => {
+        if (cellGroupType === CellGroupType.FIXED) {
+          return columnIndex;
+        }
+        if (cellGroupType === CellGroupType.SCROLLABLE) {
+          return columnIndex - fixedColumns.length;
+        } else if (cellGroupType === CellGroupType.FIXED_RIGHT) {
+          return columnIndex - fixedColumns.length - scrollableColumns.length;
+        }
+      };
+
+      const _getColumnGroupLocalIndex = (columnGroupIndex, cellGroupType) => {
+        if (cellGroupType === CellGroupType.FIXED) {
+          return columnGroupIndex;
+        }
+        if (cellGroupType === CellGroupType.SCROLLABLE) {
+          return columnGroupIndex - fixedColumnGroups.length;
+        } else if (cellGroupType === CellGroupType.FIXED_RIGHT) {
+          return (
+            columnGroupIndex -
+            fixedColumnGroups.length -
+            scrollableColumnGroups.length
+          );
+        }
+      };
+
       const getCellGroupWidth = (cellGroupType = CellGroupType.SCROLLABLE) => {
         const container = _getColumnContainerByCellGroupType(cellGroupType);
         return container.reduce((sum, column) => sum + column.width, 0);
       };
 
-      const getColumn = (
-        columnIndex,
-        cellGroupType = CellGroupType.SCROLLABLE
-      ) => {
+      const _getColumn = (localColumnIndex, cellGroupType) => {
         const container = _getColumnContainerByCellGroupType(cellGroupType);
-        return _getMinimalColumn(container[columnIndex]);
+        return _getMinimalColumn(container[localColumnIndex]);
+      };
+
+      const getColumn = (columnIndex) => {
+        const cellGroupType = _getCellGroupTypeFromColumnIndex(columnIndex);
+        const localColumnIndex = _getColumnLocalIndex(
+          columnIndex,
+          cellGroupType
+        );
+        return _getColumn(localColumnIndex, cellGroupType);
       };
 
-      const getColumnCount = (cellGroupType = CellGroupType.SCROLLABLE) => {
+      const getColumnCount = (cellGroupType = null) => {
+        if (isNil(cellGroupType)) {
```

Comment:
Nice, I like this! 
This lets the user query columns for each cell group, and also for the whole table.

Let's make sure to add this to the docs -- #651)